### PR TITLE
BUGFIX: Multitool rewrite InsertOptionalData operations fail if run.artifacts refered.

### PR DESCRIPTION
**Pull request by harleenkohli**
if referenced run.artifacts using the index property.

### CODE REVIEW

**Comment by harleenkohli** [comment]
>             RunTest("CoreTests.sarif", 

i did not use Coretests file because it s 2.0.0 - i could have upgraded it but didnt want to mess with existing cases. Hence created a new input file with "_rtm" that has run.artifacts

---
Refers to: src/Test.UnitTests.Sarif/Visitors/InsertOptionalDataVisitorTests.cs:125 in c6a7ff4. [](commit_id = c6a7ff40b42505e2e6aee11ca53e976051bf5683, deletion_comment = False)

File: src/ReleaseHistory.md [hunk 1]
```diff
@@ -412,4 +412,7 @@
 ## **v2.1.6** [Sdk](https://www.nuget.org/packages/Sarif.Sdk/2.1.6) | [Driver](https://www.nuget.org/packages/Sarif.Driver/2.1.6) | [Converters](https://www.nuget.org/packages/Sarif.Converters/2.1.6) | [Multitool](https://www.nuget.org/packages/Sarif.Multitool/2.1.6)
 * BUGFIX: Fortify FPR converter does not populate originalUriBaseIds if the source is a drive letter (e.g. C:)
 * BUGFIX: Multitool rebaseUri command throws null reference exception if results reference run.artifacts using the index property.
-* BUGFIX: Pre-release transformer does not upgrade schema uri if input version is higher than rtm.1.
\ No newline at end of file
+* BUGFIX: Pre-release transformer does not upgrade schema uri if input version is higher than rtm.1.
+
+## **v2.1.7** [Sdk](https://www.nuget.org/packages/Sarif.Sdk/2.1.7) | [Driver](https://www.nuget.org/packages/Sarif.Driver/2.1.7) | [Converters](https://www.nuget.org/packages/Sarif.Converters/2.1.7) | [Multitool](https://www.nuget.org/packages/Sarif.Multitool/2.1.7)
+* BUGFIX: Multitool rewrite InsertOptionalData operations fail if results reference run.artifacts using the index property.
```
Comment (michaelcfanning) [inline src/ReleaseHistory.md]: >run.arti [](start = 84, length = 8)

Can we please retain the discipline of using markdown backticks to encapsulate reference to properties? Open this file in any markdown viewer to see how these new items compare to previous,

`run.artifacts` 

there are other instances of this NIT in this file.

**Comment by michaelcfanning** [comment]
>             RunTest("CoreTests.sarif", 

Why not update the previous tests? Now we'll have to go eliminate these files. :)

---
In reply to: [506919398](https://github.com/microsoft/sarif-sdk/pull/1551#issuecomment-506919398) [](ancestors = 506919398)

---
Refers to: src/Test.UnitTests.Sarif/Visitors/InsertOptionalDataVisitorTests.cs:125 in c6a7ff4. [](commit_id = c6a7ff40b42505e2e6aee11ca53e976051bf5683, deletion_comment = False)

**Comment by harleenkohli** [comment]
>             RunTest("CoreTests.sarif", 

Mike, sorry i may have misunderstood your comment. I will provide my inputs on the basis of my interpretation of the above comment: 

The 2.0.0 tests are still valid and do not need to be eliminated - these go through the path of first transforming to latest version and then applying Optional data. I think its a good test to keep.

generally - i like the fact that we have a variety of input versions in our test cases - they all use our transformer internally to upgrade before testing the "actual" thing they are supposed to be testing. so this is a bonus verification that these tests do - i have often benefitted from these tests while making changes in the schema. I think we should keep them :)

---
In reply to: [507405134](https://github.com/microsoft/sarif-sdk/pull/1551#issuecomment-507405134) [](ancestors = 507405134,506919398)

---
Refers to: src/Test.UnitTests.Sarif/Visitors/InsertOptionalDataVisitorTests.cs:125 in c6a7ff4. [](commit_id = c6a7ff40b42505e2e6aee11ca53e976051bf5683, deletion_comment = False)

File: src/ReleaseHistory.md (hunk 1 above)
Comment (harleenkohli) [reply src/ReleaseHistory.md]: thanks! noted ! :) i will update

---
In reply to: [299195692](https://github.com/microsoft/sarif-sdk/pull/1551#discussion_r299195692) [](ancestors = 299195692)

**Comment by harleenkohli** [comment]
>             RunTest("CoreTests.sarif", 

based on offline discussion - i will upgrade the 2.0.0s

---
In reply to: [507446539](https://github.com/microsoft/sarif-sdk/pull/1551#issuecomment-507446539) [](ancestors = 507446539,507405134,506919398)

---
Refers to: src/Test.UnitTests.Sarif/Visitors/InsertOptionalDataVisitorTests.cs:125 in c6a7ff4. [](commit_id = c6a7ff40b42505e2e6aee11ca53e976051bf5683, deletion_comment = False)

**Pull request merged**